pyrobosim/pyrobosim/gui/main.py

```diff
@@ -306,3 +311,11 @@ def on_close_click(self):
             print(f"[{robot.name}] Closing {robot.location}")
             self.canvas.close_location(robot)
             self.update_button_state()
+
+    def on_toggle_click(self):
+        """Callback to toggle collision polygons."""
+        robot = self.get_current_robot()
+        if robot and robot.location:
+            print(f"[{robot.name}] Toggling collision polygons")
+            self.canvas.toggle_collision_polygons()
+            self.update_button_state()
```

Comment:
Instead of this line, which does a bunch of other stuff that isn't needed, you can instead do:

```python
self.canvas.draw_signal.emit()
```